src/cocotb/types/__init__.py

```diff
@@ -4,7 +4,22 @@
 from ._abstract_array import AbstractArray
 from ._array import Array
 from ._logic import Logic
-from ._logic_array import LogicArray
+from ._logic_array import RESOLVE_X, LogicArray, ResolveX
 from ._range import Range
 
-__all__ = ("Range", "Array", "Logic", "LogicArray", "AbstractArray")
+__all__ = (
+    "Range",
+    "Array",
+    "Logic",
+    "LogicArray",
+    "ResolveX",
+    "RESOLVE_X",
+    "AbstractArray",
+)
+
+# Change module to get correct linking in Sphinx
+AbstractArray.__module__ = __name__
+Array.__module__ = __name__
+LogicArray.__module__ = __name__
+Range.__module__ = __name__
+ResolveX.__module__ = __name__
```

Comment:
See if this is still necessary. This is pretty ugly and not necessary in other places where we re-export things.

Reply:
Still required. Problem is that the code examples otherwise try to link to `cocotb.types._range.Range` etc. which doesn't have documentation.

Reply:
This is probably also why `start_soon` and a couple other things don't link. I think we need a solution (put in `cocotb.utils`) that handles re-exports properly. The `atpublic` library looks nice, but it's a pretty minor thing to add our first dependency for.

Reply:
Maybe pick this in? https://github.com/ktbarrett/cocotb/commit/7d6b1d9f5bef384e4c8ade9b3a8d98e229673dc0.